### ISSUE TITLE
Intent classification using phonemes sequence

### ISSUE THREAD

**Issue by ellurunaresh** [ajinkyaT/CNN_Intent_Classification#3]
Thanks for sharing the code. 
I was trying to do intent classification using phoneme sequence instead of the word-based approach.  
Instead of word embeddings from glove I was using integer phoneme indexes. By somehow my model is performing overfitting. I have tried regularizers to prevent but it couldn't help.
Do you have any idea how to solve this problem?

Train set: (21265, 100) (21265, 7)
validation set: (2362, 100) (2362, 7)

Epoch 1/20
21265/21265 [==============================] - 16s 732us/step - loss: 0.0954 - acc: 0.9876 - val_loss: 4.4168 - val_acc: 0.5881
Epoch 2/20
21265/21265 [==============================] - 16s 733us/step - loss: 0.0854 - acc: 0.9903 - val_loss: 4.4084 - val_acc: 0.5881
Epoch 3/20
21265/21265 [==============================] - 16s 731us/step - loss: 0.0854 - acc: 0.9895 - val_loss: 4.6018 - val_acc: 0.5910
Epoch 4/20
21265/21265 [==============================] - 15s 718us/step - loss: 0.0867 - acc: 0.9896 - val_loss: 4.4348 - val_acc: 0.5893
Epoch 5/20
21265/21265 [==============================] - 15s 722us/step - loss: 0.0815 - acc: 0.9902 - val_loss: 4.6274 - val_acc: 0.5876
Epoch 6/20
21265/21265 [==============================] - 16s 734us/step - loss: 0.0797 - acc: 0.9905 - val_loss: 4.8017 - val_acc: 0.5876
Epoch 7/20
21265/21265 [==============================] - 16s 750us/step - loss: 0.0823 - acc: 0.9901 - val_loss: 4.7837 - val_acc: 0.5881
Epoch 8/20
21265/21265 [==============================] - 16s 729us/step - loss: 0.0769 - acc: 0.9904 - val_loss: 5.0681 - val_acc: 0.5876
Epoch 9/20
21265/21265 [==============================] - 15s 729us/step - loss: 0.0769 - acc: 0.9909 - val_loss: 4.7879 - val_acc: 0.5881
Epoch 10/20
21265/21265 [==============================] - 16s 760us/step - loss: 0.0763 - acc: 0.9905 - val_loss: 4.7429 - val_acc: 0.5881

**Comment by ajinkyaT** [comment]
My guess would be:

1) Words in the test dataset may not be available for training in the training dataset
2) Check for grammar, are the words lemmatized? lowercased? etc. 
3) I am a bit sceptical for using integer-based phonemes for training as the advantage of using a vector-based model with the embedding layer is to map semantically similar words close to each other in the latent space, this vector representation is then fed into CNN. It would be better to use LSTM or RNN in case of phonemes based methods as those are well suited for sequence data which is the case with you.

**Comment by ellurunaresh** [comment]
Thanks for the reply.
1.  Yes, the validation was unseen in the training. The full set was divided into train and validation set.
2.  All the words are in lower case and made into a phoneme sequence
3.  Yes. I have tried with simple BiLSTM but didn't converge. Can we do vector representation using the phonemes (integers)

**Comment by ajinkyaT** [comment]
Sorry, but I am not aware of any vector representation technique using phonemes.

**Comment by ajinkyaT** [comment]
You may have a look at this article for more clarity, [NLP Learning Series: Part 3 - Attention, CNN and what not for Text Classification](https://mlwhiz.com/blog/2019/03/09/deeplearning_architectures_text_classification/)

**Comment by ellurunaresh** [comment]
Sorry for the late reply
I will go through the link.
